ScanShop.Mobile/ScanShop.Mobile/App.xaml.cs

```diff
@@ -1,20 +1,30 @@
 ﻿using ScanShop.Mobile.Services;
-using ScanShop.Mobile.Views;
-using System;
+using Xamarin.Essentials;
 using Xamarin.Forms;
-using Xamarin.Forms.Xaml;
 
 namespace ScanShop.Mobile
 {
     public partial class App : Application
     {
-
         public App()
         {
             InitializeComponent();
 
             DependencyService.Register<MockDataStore>();
+
             MainPage = new AppShell();
+
+            if (string.IsNullOrEmpty(SecureStorage.GetAsync("BearerToken").Result))
+            {
+                Shell.Current.GoToAsync("//LoginPage");
+            }
+            else
+            {
+                var bearerToken = SecureStorage.GetAsync("BearerToken");
```

Comment:
firstly, you call SecureStorage.GetAsync("BearerToken"); twice. Secondly, I think it is better to brought setting bearer into httpClientService, for example in InitializeAsync method with injecting IConfiguration there

Reply:
Fixed in new commit